### ISSUE TITLE
MessageSearchQuery for link attachments without preview images

### ISSUE THREAD

**Issue by saidReclip** [GetStream/stream-chat-swift#2363]
## What are you trying to achieve?
To get MessageSearchQuery search for messages with link attachments without preview image.

## If possible, how can you achieve this currently?
Right now it is only possible to search for link attachments with preview.

**Comment by saidReclip** [comment]
Also reporting that not all links with preview images are showing up in MessageSearchQuery list. For example "https://www.youtube.com/watch?v=jfKfPfyJRdk" or any other YouTube video's link.

**Comment by nuno-vieira** [comment]
Hi @saidReclip,

Can you please provide more details? Can you show a snippet of what you are trying to do?

Best,
Nuno

**Comment by saidReclip** [comment]
@nuno-vieira 
```
        var attachments: Set<AttachmentType> = []

            attachments = [.image, .linkPreview]

        let query = MessageSearchQuery(
            channelFilter: .equal(.cid, to: channel.cid),
            messageFilter: .withAttachments(attachments),
            sort: [.init(key: .createdAt, isAscending: false)],
            pageSize: messagesPageSize
        )

        messageSearchController.search(query: query, completion: { [weak self] _ in
            guard let self = self else { return }
            self.updateAttachments()
        })

```

**Comment by saidReclip** [comment]
@nuno-vieira  I am trying to get a list of links sent in the channel, regardless it has preview image or has not. 
This code is expected to search for messages that contains links in them. But currently it skips some of the links. (examples above)

**Comment by saidReclip** [comment]
Hey @nuno-vieira, after several tries I noticed that if I expand the list of attachments to `[.image, .video, .linkPreview, .file, .giphy]` it will work well for YouTube video links. Do you have docs explaining this logic? 
Nonetheless we still have a problem for links that does not have any preview images.
<img width="191" alt="Снимок экрана 2022-11-07 в 12 48 49" src="https://user-images.githubusercontent.com/102264702/200279684-bed18175-ff49-4322-a927-4511ba32cd85.png">

**Comment by nuno-vieira** [comment]
Hi @saidReclip,

I just tested this in our DemoApp, and it looks like link-only attachments are missing the `type: link` property, which is most likely the issue. I will get in contact with the Backend and will let you know once we have more details.

Best,
Nuno

**Comment by nuno-vieira** [comment]
Hi @saidReclip,

Actually, this is an issue in the iOS SDK. There are no such things as "link" attachments. So if you want to get all the messages that have attachments, this is the right way:
```swift
messageFilter: .and([
    .exists(FilterKey<MessageSearchFilterScope, String>(stringLiteral: "attachments"))
])
```

This is not very friendly at the moment and is also not documented, so we will definitely improve this. But for now, this should unblock you. 

Best,
Nuno

**Comment by polqf** [comment]
Hi! This has been fixed in our latest release [4.25.0](https://github.com/GetStream/stream-chat-swift/releases/tag/4.25.0)
Closing this issue for now, feel free to reopen it if you encounter any issues.
